integration/pvtdatapurge/data_purge_test.go

```diff
@@ -301,136 +254,106 @@ var _ = Describe("Pvtdata purge", func() {
+			marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-8-keep", "color":"green", "size":42, "owner":"simon", "price":180}`, org2Peer0)
+			marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-9-purge", "color":"red", "size":24, "owner":"heather", "price":635}`, org2Peer0)
+			marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-9-purge"}`, org2Peer0)
+
+			// stop original peers
+			stopPeer(network, processes, org1Peer0)
+			stopPeer(network, processes, org2Peer0)
+			stopPeer(network, processes, org3Peer0)
+			network.Peers = []*nwo.Peer{}
 
-		It("should enable successful peer reconciliation with partial write-sets", func() {
-			marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-1", "color":"green", "size":42, "owner":"simon", "price":180}`, org2Peer0)
-			marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-2", "color":"red", "size":24, "owner":"heather", "price":635}`, org2Peer0)
-			marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-3", "color":"black", "size":12, "owner":"bob", "price":2}`, org2Peer0)
+			// restart new peer by itself and wait for "Could not fetch" message
+			startPeer(network, processes, peerRunners, org3Peer1)
+			network.Peers = append(network.Peers, org3Peer1)
+			runner := peerRunners[org3Peer1.ID()]
+			Eventually(runner.Err(), network.EventuallyTimeout).Should(gbytes.Say("Could not fetch \\(or mark to reconcile later\\) 2 eligible collection private write sets for block \\[\\d+\\] \\(0 from local cache, 0 from transient store, 0 from other peers\\)\\. Will commit block with missing private write sets:\\[txID: [0123456789abcdef]+, seq: \\d+, namespace: marblesp, collection: collectionMarblePrivateDetails"))
 
-			marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-1"}`, org2Peer0)
-			marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-3"}`, org2Peer0)
+			// restart original peers
+			startPeer(network, processes, peerRunners, org1Peer0)
+			network.Peers = append(network.Peers, org1Peer0)
 
-			ledgerHeight := nwo.GetLedgerHeight(network, network.Peers[0], channelID)
+			startPeer(network, processes, peerRunners, org2Peer0)
+			network.Peers = append(network.Peers, org2Peer0)
 
-			for _, peer := range network.Peers {
-				stopPeer(network, processes, peer)
-			}
-			stoppedPeers := network.Peers
-			network.Peers = []*nwo.Peer{}
+			startPeer(network, processes, peerRunners, org3Peer0)
+			network.Peers = append(network.Peers, org3Peer0)
 
-			startNewPeer(network, orderer, org3Peer1, ledgerHeight, processes, peerRunners)
+			// Wait for reconciliation to complete and then check for reconciled data
+			time.Sleep(30 * time.Second)
```

Comment:
Did you need to have this fixed timeout here? The fixed timeout always end up adding unnecessary delays. The function `AssertPresentInCollectionM` underneath uses `Eventually` mechanism with a default timeout. If we need a higher than the default timeout here, my suggestion would be to introduce a new parameter to this function and pass down the timeout.